DFC.ServiceTaxonomy.GraphSync/Neo4j/Queries/NodeAndIncomingRelationshipsQuery.cs

```diff
@@ -41,11 +43,21 @@ public Query Query
         {
             get
             {
-                var commandStringBuilder = new StringBuilder($"match (s)-[r]->(d:{string.Join(":", _sourceNodeLabels)} {{{_sourceNodePropertyIdName}: '{_sourceNodeId}'}})");
-                commandStringBuilder.AppendLine(" with s, {destNode: d, relationship: r, destinationIncomingRelationships:collect({destIncomingRelationship:'todo',  destIncomingRelSource:'todo'})} as relationshipDetails");
-                commandStringBuilder.AppendLine(" with { sourceNode: s, outgoingRelationships: collect(relationshipDetails)} as nodeAndOutRelationshipsAndTheirInRelationships");
-                commandStringBuilder.AppendLine(" return nodeAndOutRelationshipsAndTheirInRelationships");
-                return new Query(commandStringBuilder.ToString());
+                if (!_ignoreIncomingRelationship)
+                {
+                    var commandStringBuilder = new StringBuilder($"match (s)-[r]->(d:{string.Join(":", _sourceNodeLabels)} {{{_sourceNodePropertyIdName}: '{_sourceNodeId}'}})");
+                    commandStringBuilder.AppendLine(" with s, {destNode: d, relationship: r, destinationIncomingRelationships:collect({destIncomingRelationship:'',  destIncomingRelSource:'todo'})} as relationshipDetails");
+                    commandStringBuilder.AppendLine(" with { sourceNode: s, outgoingRelationships: collect(relationshipDetails)} as nodeAndOutRelationshipsAndTheirInRelationships");
+                    commandStringBuilder.AppendLine(" return nodeAndOutRelationshipsAndTheirInRelationships");
+                    return new Query(commandStringBuilder.ToString());
+                }
+                else
+                {
+                    var commandStringBuilder = new StringBuilder($"match (s:{string.Join(":", _sourceNodeLabels)} {{{_sourceNodePropertyIdName}: '{_sourceNodeId}'}})");
```

Comment:
Can we use an existing query instead of adding the flag to this one (perhaps `NodeWithOutgoingRelationshipsQuery`)?

At some point, I think we need to consolidate/rationalise the queries we have (although not needed for this PR!).

Reply:
Yeah it felt a bit dirty either way, but I'll move it into the other query 👍